libraries/Microsoft.Bot.Builder.Core.Extensions/FileTranscriptStore.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    /// <summary>
+    /// The file transcript store stores transcripts in file system with each activity as a file.
+    /// </summary>
+    public class FileTranscriptStore : ITranscriptStore
+    {
+        private static JsonSerializerSettings jsonSettings = new JsonSerializerSettings()
+        {
+            Formatting = Formatting.Indented,
+            NullValueHandling = NullValueHandling.Ignore
+        };
+        protected string rootFolder;
+
+        public FileTranscriptStore(string folder)
+        {
+            this.rootFolder = folder ?? throw new ArgumentNullException("Missing folder");
+        }
+
+        /// <summary>
+        /// Log an activity to the transcript
+        /// </summary>
+        /// <param name="activity">activity to log</param>
+        /// <returns></returns>
+        public async Task LogActivity(IActivity activity)
+        {
+            if (activity == null)
```

Comment:
You can use the BotAssert class for this. We've got assertions for all the common Bot realted null checks in there.